public/app/features/alerting/unified/state/actions.ts

```diff
@@ -366,3 +372,17 @@ export const expireSilenceAction = (alertManagerSourceName: string, silenceId: s
     dispatch(fetchAmAlertsAction(alertManagerSourceName));
   };
 };
+
+export const createSilence = async (
```

Comment:
I suggest to use the same pattern as with `updateAlertManagerConfigAction` , using `createAsyncThunk` to have request state in history

Reply:
Good point and it also provides a `loading` and `error`